pkg/apis/resource/validation/validation.go

```diff
@@ -738,16 +742,37 @@ func validateRawExtension(rawExtension runtime.RawExtension, fldPath *field.Path
 	return allErrs
 }
 
+const interfaceNameMaxLength int = 256
+const hardwareAddressMaxLength int = 128
+
 func validateNetworkDeviceData(networkDeviceData *resource.NetworkDeviceData, fldPath *field.Path) field.ErrorList {
 	var allErrs field.ErrorList
 	if networkDeviceData == nil {
 		return allErrs
 	}
-	allErrs = append(allErrs, validateSlice(networkDeviceData.Addresses, -1,
+
+	if len(networkDeviceData.InterfaceName) > interfaceNameMaxLength {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("InterfaceName"), networkDeviceData.InterfaceName, validation.MaxLenError(interfaceNameMaxLength)))
+	}
+
+	if len(networkDeviceData.HardwareAddress) > hardwareAddressMaxLength {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("HardwareAddress"), networkDeviceData.HardwareAddress, validation.MaxLenError(hardwareAddressMaxLength)))
```

Comment:
use `field.TooLong(...`

Note that `HardwareAddress` -> `hardwareAddress`

Reply:
Still asking for TooLong()

Reply:
Fixed, thank you